src/Controller/Controller.php

```diff
@@ -572,6 +572,9 @@ public function render($view = null, $layout = null)
             return $this->response;
         }
 
+        // Re-fetch View class to pass view variables set in beforeRender callbacks
+        $this->View = $this->getView();
+
         if ($viewClass !== $this->viewClass) {
             $this->View = $this->getView($this->viewClass);
```

Comment:
@lorenzo Yes. I don't think one shouldn't be configuring a view before deciding which view class should be actually used.